queue/queue.go

```diff
@@ -88,6 +95,22 @@ func (cq *ConcurrentQueue) start() {
 				}
 			}
 		}
+
+		// Incoming channel has been closed. Empty overflow queue into
+		// the outgoing channel.
+		nextElement := cq.overflow.Front()
+		for nextElement != nil {
+			select {
+			case cq.chanOut <- nextElement.Value:
+				cq.overflow.Remove(nextElement)
+			case <-cq.quit:
+				return
+			}
+			nextElement = cq.overflow.Front()
+		}
+
+		// Close outgoing channel.
+		close(cq.chanOut)
```

Comment:
maybe this should be done on quit also?

Reply:
I don't think so. It would signal a clean end to the queue, which isn't the case for quit